tests/test_utils.py

```diff
@@ -0,0 +1,25 @@
+from unittest.mock import Mock
+
+import pytest
+from llm_gateway.utils import max_retries
+from openai.error import APIError
+
+
+def test_retry_decorator():
+    retry_mock = Mock()
+    retry_mock.side_effect = [APIError("test"), "success"]
```

Comment:
```suggestion
    retry_mock.side_effect = [APIError("test"), APIError("test"), "success"]
```

Originally, the second time you called it (with `matching_exception`) it would just return `"success"` the first time, so the retry logic wasn't actually being tested.